src/main/java/forms/CreateListingForm.java

```diff
@@ -1,4 +1,42 @@
 package forms;
 
-public class CreateListingForm {
+import entities.User;
+import features.Listing;
+
+public class CreateListingForm extends Form{
+
+
+    private final User seller;
+    private float price;
+    private String description;
+    private String[] images;
+
+    private int id = 0;
+    public CreateListingForm(String title, float price, User seller, String description, String[] images) {
+        super(title);
+        this.price = price;
+        this.seller = seller;
+        this.description = description;
+        this.images = images;
+    }
+
+
+    public void submitListing(){
+        // This function will be called on the "submit" button click
+        // This will add a listing to the databse using the database interactor method.
+        String title = this.getTitle();
+        Listing L = new Listing(title, id,price, seller, description, images);
+        id++;
+
+        // addListing(L);
+    }
+
+    @Override
+    protected boolean validateForm() {
+        return false;
```

Comment:
you can go ahead and add validation if you'd like. things like checking that the User actually exists, the price is positive, maybe a char limit on description or title...